pennylane/templates/embeddings/iqp.py

```diff
@@ -12,55 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 r"""
-Contains the ``IQPEmbedding`` template.
+Contains the IQPEmbedding template.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
 from itertools import combinations
 
 import pennylane as qml
-from pennylane.templates.decorator import template
-from pennylane.ops import RZ, MultiRZ, Hadamard
-from pennylane.templates import broadcast
-from pennylane.wires import Wires
+from pennylane.operation import Operation, AnyWires
 
 
-def _preprocess(features, wires, pattern):
-    """Validate and pre-process inputs as follows:
-
-    * Check that the features tensor is one-dimensional.
-    * Check that the first dimension of the features tensor
-      has length :math:`n`, where :math:`n` is the number of qubits.
-    * If pattern is None, create default pattern. Else cast wires in user-provided pattern to Wires objects.
-
-    Args:
-        features (tensor_like): input features to pre-process
-        wires (Wires): wires that template acts on
-        pattern (list[int]): specifies the wires and features of the entanglers
-
-    Returns:
-        list[Wires]: preprocessed pattern
-    """
-    shape = qml.math.shape(features)
-
-    if len(shape) != 1:
-        raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
-
-    n_features = shape[0]
-    if n_features != len(wires):
-        raise ValueError(f"Features must be of length {len(wires)}; got length {n_features}.")
-
-    if pattern is None:
-        # default is an all-to-all pattern
-        pattern = [Wires(wire_pair) for wire_pair in combinations(wires, 2)]
-    else:
-        # convert wire pairs to Wires object
-        pattern = [Wires(wire_pair) for wire_pair in pattern]
-
-    return pattern
-
-
-@template
-def IQPEmbedding(features, wires, n_repeats=1, pattern=None):
+class IQPEmbedding(Operation):
     r"""
     Encodes :math:`n` features into :math:`n` qubits using diagonal gates of an IQP circuit.
```

Comment:
what does "IQP" stand for?

Reply:
The embedding encodes into the diagonal gates of an IQP circuit...once more a case of an embedding which does not (yet) have a good name in the community